src/pybind/mgr/dashboard/controllers/osd.py

```diff
@@ -196,18 +197,23 @@ def destroy(self, svc_id):
         CephService.send_command(
             'mon', 'osd destroy-actual', id=int(svc_id), yes_i_really_mean_it=True)
 
-    @RESTController.Resource('GET')
-    def safe_to_destroy(self, svc_id):
+    @Endpoint('GET', query_params=['ids'])
+    @ReadPermission
+    def safe_to_destroy(self, ids):
         """
-        :type svc_id: int|[int]
+        :type ids: int|[int]
         """
-        if not isinstance(svc_id, list):
-            svc_id = [svc_id]
-        svc_id = list(map(str, svc_id))
+
+        ids = json.loads(ids)
+        if isinstance(ids, list):
+            ids = list(map(str, ids))
+        else:
+            ids = [str(ids)]
+
         try:
             result = CephService.send_command(
-                'mon', 'osd safe-to-destroy', ids=svc_id, target=('mgr', ''))
-            result['is_safe_to_destroy'] = set(result['safe_to_destroy']) == set(map(int, svc_id))
+                'mon', 'osd safe-to-destroy', ids=ids, target=('mgr', ''))
+            result['is_safe_to_destroy'] = set(result['safe_to_destroy']) == set(map(int, ids))
```

Comment:
The code looks very similar to the one used before. The only thing I could identify being changed here is the way multiple IDs are passed. I like the way it's done now but wonder if the old one hasn't worked?

Reply:
The old code only worked for a single OSD, if you tried to pass multiple OSD the send_command would fail, then catched by the `try` and always returned false.

Reply:
The old code only worked for single OSD.
If you passed multiple OSDs the `send_command` would fail because it didn't recognize the format of the argument. 
And since we have a try block, that failure would be catched and we would always return `false`.